ledger/participant-integration-api/src/main/scala/platform/store/backend/oracle/OracleTable.scala

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.oracle
+
+import java.sql.Connection
+import com.daml.platform.store.backend.common.{BaseTable, Field, Table}
+
+private[oracle] object OracleTable {
+  private def idempotentBatchedInsertBase[FROM](
+      insertStatement: String,
+      keyFieldIndex: Int,
+  )(fields: Seq[(String, Field[FROM, _, _])]): Table[FROM] =
+    new BaseTable[FROM](fields) {
+      override def executeUpdate: Array[Array[_]] => Connection => Unit =
+        data =>
+          connection =>
+            Table.ifNonEmpty(data) {
+              val preparedStatement = connection.prepareStatement(insertStatement)
+              data(0).indices.foreach { dataIndex =>
+                fields(keyFieldIndex)._2.prepareData(
+                  preparedStatement,
+                  1,
+                  data(keyFieldIndex)(dataIndex),
+                )
+                fields.indices.foreach { fieldIndex =>
+                  preparedStatement.setObject(fieldIndex + 2, data(fieldIndex)(dataIndex))
```

Comment:
perhaps we need here the same treatment as in the other places:
fields(fieldIndex)._2.prepareData(
                    preparedStatement,
                    fieldIndex + 2,
                    data(fieldIndex)(dataIndex),
                  )

Reply:
i think just this one place needed it

Reply:
needed it for what? this approach bypasses the path what is layed down in the OracleFields.OracleTimestamp. This means even it is a timestamp it will be passed with setObject.
This is very disturbing...why is this working at all?
This is used for package insertion, it has a timestamp, and a mandatory one.
It seems to me the setTimestamp for the timestamp insertion is not needed....
But if that it not needed, then the whole .prepareData extension to the framework is not needed neither.
Do you know what is going on here?

Reply:
ahh I might have a theory: it is working because it is never null in this case, it might be that the null timestamps are not working, and they need to be put as .setTimestamp. If that is the case I would just change here to use the .prepareData (how I suggested above), and done

Reply:
Err I don't think we wrote this code, I think you did :) It's just copy pasta from the H2 implementation - I'm not sure I entirely understand the change you are requesting, but feel free to make it for both H2 and Oracle if you think it simplifies.

Reply:
fixed